sql/src/main/java/io/crate/execution/engine/window/WindowFunctionBatchIterator.java

```diff
@@ -192,11 +227,32 @@ public boolean hasNext() {
                     pEnd = findFirstNonPeer(sortedRows, pStart, end, cmpPartitionBy);
                 }
 
+                if (windowDefinition.windowFrameDefinition().type() == RANGE) {
+                    // if the offsetCell position is -1 the window is ordered by a Literal so we leave the
```

Comment:
Would it be possible to eliminate the `order by` altogether if it is on a literal? E.g. in the analyzer parts. Or does that break the preceding/following logic altogether?